scripts/update.sh

```diff
@@ -2,32 +2,33 @@
 # shellcheck source=/dev/null
 source "/home/steam/server/helper_functions.sh"
 
-UpdateRequired
-updateRequired=$?
 # Check if Update was actually required
-if [ "$updateRequired" != 0 ]; then
-  exit 0
-fi
+if UpdateRequired; then
+    if [ "${UPDATE_ON_BOOT}" = false ]; then
+        LogWarn "An update is available however, UPDATE_ON_BOOT needs to be enabled for auto updating"
+        DiscordMessage "An update is available however, UPDATE_ON_BOOT needs to be enabled for auto updating" "warn"
+        exit 1
+    fi
 
-if [ "${UPDATE_ON_BOOT}" = false ]; then
-    LogWarn "An update is available however, UPDATE_ON_BOOT needs to be enabled for auto updating"
-    DiscordMessage "An update is available however, UPDATE_ON_BOOT needs to be enabled for auto updating" "warn"
-    exit 1
-fi
+    if [ "${RCON_ENABLED,,}" = false ]; then
+        LogWarn "An update is available however auto updating without rcon is not supported"
+        DiscordMessage "An update is available however auto updating without rcon is not supported" "warn"
+        exit 1
+    fi
 
-if [ "${RCON_ENABLED,,}" = false ]; then
-    LogWarn "An update is available however auto updating without rcon is not supported"
-    DiscordMessage "An update is available however auto updating without rcon is not supported" "warn"
-    exit 1
-fi
+    if [[ "${AUTO_UPDATE_WARN_MINUTES}" =~ ^[0-9]+$ ]]; then
+        DiscordMessage "Server will update in ${AUTO_UPDATE_WARN_MINUTES} minutes"
+    fi
 
-if [ "$(get_player_count)" -gt 0 ]; then
-  LogAction "Updating the server from $CURRENT_MANIFEST to $TARGET_MANIFEST."
-  DiscordMessage "Server will update in ${AUTO_UPDATE_WARN_MINUTES} minutes"
-  RCON "broadcast The_Server_will_update_in_${AUTO_UPDATE_WARN_MINUTES}_Minutes"
-  sleep "${AUTO_UPDATE_WARN_MINUTES}m"
-fi
+    if countdown_message "${AUTO_UPDATE_WARN_MINUTES}" "Server_will_update"; then
+        LogAction "Updating the server from $CURRENT_MANIFEST to $TARGET_MANIFEST."
+        rm /palworld/steamapps/appmanifest_2394010.acf
 
-rm /palworld/steamapps/appmanifest_2394010.acf
-backup
-RCON "shutdown 1"
\ No newline at end of file
+        backup
+        shutdown_server
+    elif [ -z "${AUTO_UPDATE_WARN_MINUTES}" ]; then
+        LogWarn "Unable to auto update, the server is not empty and AUTO_UPDATE_WARN_MINUTES is empty."
+    else
+        LogWarn "Unable to auto update, the server is not empty and AUTO_UPDATE_WARN_MINUTES is not an integer: ${AUTO_UPDATE_WARN_MINUTES}"
```

Comment:
Can be removed if the check for an Integer is already made above.

Reply:
No, the integer check is different from this.

The integer checks only happens if there's an issue with the countdown, since the countdown is set to not check the timer if there are no players the second elif only happens if there are players and the countdown is empty or is not an integer.

The two LogWarn give you different information.